### PR TITLE
Feat/chat plugin use render library

### LINKED ISSUE #820: Use RenderLibrary in chatplugin
In chatplugin we use preact and to use the RenderLibrary we need to make the chatplugin compatible for react.

### PR DESCRIPTION
**Pull request by chrismatix**
fixes #820 

Notable changes:
- Use `preact/compat` to make the chat plugin compatible with React libraries
- Remove the router form the chat plugin
- Replace `AiryMessage` with the render library
- Improve the props API of `SourceMessage`

### CODE REVIEW

File: frontend/chat-plugin/bazel.tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,14 @@
 {
   "extends": "../../bazel.tsconfig.json",
   "compilerOptions": {
-    "jsxFactory": "h"
+    "skipLibCheck": true,
+    "paths": {
+      "react": [
+        "preact/compat"
```
Comment (chrismatix) [inline frontend/chat-plugin/bazel.tsconfig.json:7]: This aliasing is picked up by our webpack rules

File: lib/typescript/render/shared.ts [hunk 2]
```diff
@@ -1,20 +1,19 @@
-import {Conversation, isFromContact, Message} from 'httpclient';
+import {isFromContact, Message, Contact} from 'httpclient';
 import {formatTime} from 'dates';
 import {DefaultMessageRenderingProps} from './components';
 
 export interface MessageRenderProps {
   message: Message;
-  conversation: Conversation;
-  prevWasContact: boolean;
-  nextIsSameUser: boolean;
```
Comment (chrismatix) [inline lib/typescript/render/shared.ts:9]: Here is the change to the `SourceMessage` props

**Review by AitorAlgorta — APPROVED** [review]
I am curious to know if this affects the bundle size

**Comment by chrismatix** [comment]
@AitorAlgorta Excellent question. Will check that out tomorrow and do a bit more testing.

**Review by lucapette — DISMISSED** [review]
:shipit:

**Review by lucapette — DISMISSED** [review]

**Review by AitorAlgorta — DISMISSED** [review]

**Review by AitorAlgorta — APPROVED** [review]

**Pull request merged**